oslo_messaging/server.py

```diff
@@ -329,6 +342,11 @@ def __init__(self, transport, dispatcher, executor='blocking'):
         """
         self.conf = transport.conf
         self.conf.register_opts(_pool_opts)
+        self.conf.register_opts(_metrics_opts)
+
+        if self.conf.statsd_enabled:
+            self.metrics = metrics.Metrics(self.conf.statsd_host, self.conf.statsd_port)
+            self.metrics.start()
```

Comment:
We could™ call `start()` in `Metrics`'s `__init__()` unless you see a reason to call this explicitly.

Reply:
i would call the start method within the server.start():
https://github.com/sapcc/oslo.messaging/blob/stable/rocky-m3/oslo_messaging/server.py#L384

I am not sure however where to call stop. it could be within the server.stop():
https://github.com/sapcc/oslo.messaging/blob/stable/rocky-m3/oslo_messaging/server.py#L421
or within server.wait():
https://github.com/sapcc/oslo.messaging/blob/stable/rocky-m3/oslo_messaging/server.py#L421

If it is called within server.stop() it would miss messages that are still being processed. If it would be called at server.wait() it would catch all messages, but not all might call this method after stop?!